src/main/java/com/google/sps/servlets/CheckPermissionsServlet.java

```diff
@@ -80,14 +83,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
       TestIamPermissionsResponse permissionsResponse = testingPermissionsResponse.execute();
 
-      System.out.println(permissionsResponse);
+      //check if any permissions have been returned
+      if (permissionsResponse.getPermissions() == null){
+        response.getWriter().println(gson.toJson("No permissions have been assigned to the service account."));
+        return;
+      }
+
+      Boolean arePermissionsCorrect = false;
+
+      int missing = requiredPermissions.size() - permissionsResponse.getPermissions().size();
+
+      if (missing == 0) {
+        arePermissionsCorrect = true;
+      }
```

Comment:
Sam thing here
```
arePermissionsCorrect = missing = 0
```

Reply:
I had just noticed that and changed it (3f49f24) - it definitely makes it a lot simpler. Thank you for pointing out it could be done that way - I just need to remember to use it from now on!